baybe/constraints/base.py

```diff
@@ -92,6 +88,10 @@ class DiscreteConstraint(Constraint, ABC):
     eval_during_modeling: ClassVar[bool] = False
     # See base class.
 
+    _numerical_only: ClassVar[bool]
```

Comment:
Thanks! 
- Good point on setting the default `_numerical_only = False`. I will do the change.   
- Regarding adding `_numerical_only` to `Constraint` or `DiscreteConstraint `, my concern was all `ContinuousConstraint` and `CardinalityConstraint` are only valid for numerical parameters, so it might to be a bit redundant to add `_numerical_only = True` to them. Therefore, `_numerical_only` was kept to the level of `DiscreteConstraint`. Nevertheless, `_numerical_only` is actually a property of any constraint. Please let me know your preference.

Reply:
@Scienfitz: what would we gain if we moved `_numerical_only` to the `Constraint` class instead just having it in `DiscreteConstraint`?

Reply:
because
- conceptually it describes a property of any parameter, not just a discrete one
- I did not see any downside, but do we see any significant sisues? The potential redundance is tiny, like adding a line `_numerical_only=True` to the conti constraint class. I'd happily trade that so any code viewer sees that these constraints are numerical only, even if current implementations of conti parameters would anyway imply that

Reply:
Ok thanks. Will close this thread now and do my review. If I have concerns, I'll reopen again

Reply:
Reopening indeed :D 
* @Scienfitz: I guess you'd also prefer to have the default `False` being set in the abstract class?
* @Waschenbacher: note that if you don't assign it any value, the class variable is actually not created (i.e. think about what having nothing on the rhs of the equation would even mean in terms of the created class/object)
* @Scienfitz: perhaps that's also something we should revisit for the other class vars when refactoring the module

Reply:
yes the default belongs to `Constraint`

Reply:
also, why is it private? Properties like `is_numeric` of Parameters are also public

Reply:
Clear! I will set the default for `Constraint`. I make it private since I'm not sure whether we want to have it for long-term.
Q1: Would you prefer private or public? @Scienfitz @AdrianSosic 
Q2: Are you satisfied with the name `numerical_only` (either private or public)? Or `is_numeric` for consistency?

Reply:
Q1: I'd actually prefer to keep this private for now. Reason being: the variable is currently used internally for validation purposes; however, it has not yet been decided how a general validation/integration check will look like for all our modules. I'm currently in the process of collecting cases that need to be checked, and then we need to come up with a mechanism that can handle all cases. That is, I could image to have something like a `supports` method, that takes the context as input and either spits out a bool or directly throws the corresponding exceptions. For example, the corresponding method of `SearchSpace` would take the constraints, parameters, etc and can then decide based on those whether that combination is valid. And until we've decided, I'd like to avoid breaking changes.

Q2: No strong opinion here, especially when it stays private. Perhaps `numerical_only` is a bit more precise because makes it clearer that the statement is about the parameters, not the constraint itself? Don't know...

Reply:
re Q1: the same would be true for `is_numeric` in parameters and other class vars we already have and so on where these arguments have not been made to keep them private, to me this is simply a useful property of the constraint, no need to keep it private (args about future usage don't really change that)

re Q2: when we say consistency then it needs to be set to whatever has been used in other places most frequently, so its not really a preference. If no common pattern is evident, then we need to decide one, in which case its fully up to you because I don't mind either

Reply:
Thanks for the inputs. I'll go with the name `numerical_only` and make it public, unless there's a strong reason not to.